src/groupeddataframe/splitapplycombine.jl

```diff
@@ -962,14 +1056,51 @@ function _combine(f::AbstractVector{<:Pair},
                 throw(ArgumentError("a single value or vector result is required when passing " *
                                     "multiple functions (got $(typeof(res)))"))
             end
+            if !(firstres isa AbstractVector) && isnothing(idx_agg)
```

Comment:
No - exactly the opposite. This is to make sure we compute `idx_agg` exactly once.

In short we say if `firstres` is a vector do not compute `idx_agg` or do not compute it if it was already computed (as `idx_agg` is identical for each function returning single-row-per-group).

Later we use `=== idx_agg` to identify columns that need to be pseudo-broadcasted (because they were produced by single-row-per-group return value type).

I leave this unresolved as this is a crucial part of the change I have implemented so it is crucial that we are on the same page here.

Reply:
added